src/zcash/Note.cpp

```diff
@@ -226,6 +249,22 @@ boost::optional<SaplingNotePlaintext> SaplingNotePlaintext::decrypt(
     SaplingNotePlaintext ret;
     ss >> ret;
 
+    uint256 cmu_expected;
+    if (!librustzcash_sapling_compute_cm(
+        ret.d.data(),
+        pk_d.begin(),
+        ret.value(),
+        ret.rcm.begin(),
+        cmu_expected.begin()
+    ))
+    {
+        return boost::none;
+    }
+
+    if (cmu_expected != cmu) {
+        return boost::none;
+    }
+
     assert(ss.size() == 0);
```

Comment:
Here we assert after calling ```librustzcash_sapling_compute_cm```. Given that these two methods are essentially the same . It doesn't make sense to me that we are doing this two different ways. Also, the decrypt methods could be refactored to eliminate some duplication, but this can be done later.